specification/ai/data-plane/OpenAI.Assistants/OpenApiV2/preview/2024-02-15-preview/assistants_generated.json

```diff
@@ -897,9 +1000,8 @@
             "type": "string"
           },
           {
-            "name": "createRunOptions",
+            "name": "body",
```

Comment:
You should be able to avoid this by using @bodyRoot if that is what you want...

Reply:
Actually, I think we originally used `@bodyRoot` not aware that `"name"` will be changed. The only reason it was originally introduced was (if I remember correctly) so we could correctly decorate with `@added` the model. I don't think that the [swagger provided](https://github.com/Azure/azure-rest-api-specs/blob/00a6d4f6cdfe2e3ffd069d0771ee0a52143a13c0/specification/cognitiveservices/data-plane/AzureOpenAI/inference/preview/2024-05-01-preview/inference.json#L1458-L1513) by the service team modifies `"name"` at all, so we may be adding something unnecessarily here.

Reply:
I ended up using `BodyParameter<...ThreadMessageOption>` to match what we do for other operations as well.